Framework/Core/src/CommonDataProcessors.cxx

```diff
@@ -216,38 +216,41 @@ DataProcessorSpec
     int ntfmerge = 1;
 
     // values from json
-    auto fnjson = ic.options().get<std::string>("json-file");
-    if (!fnjson.empty()) {
-      auto [fnb, fmo, ntfm] = dod->readJson(fnjson);
-      if (!fnb.empty()) {
-        fnbase = fnb;
-      }
-      if (!fmo.empty()) {
-        filemode = fmo;
-      }
-      if (ntfm > 0) {
-        ntfmerge = ntfm;
+    std::string fnb, fmo;
+    int ntfm = 0;
+    if (ic.options().isSet("json-file")) {
+      auto fnjson = ic.options().get<std::string>("json-file");
+      if (!fnjson.empty()) {
+        tie(fnb, fmo, ntfm) = dod->readJson(fnjson);
```

Comment:
`auto [fnb, fmo, ntfm] = `

Reply:
true, have removed auto for some reason earlier, but can put it back again.